jupyter_client/tests/conftest.py

```diff
@@ -7,9 +7,31 @@
 
 from .utils import test_env
 
+try:
+    import resource
+except ImportError:
+    # Windows
+    resource = None
+
 pjoin = os.path.join
 
 
+# Handle resource limit
+if resource is not None:
+    soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+
+    DEFAULT_SOFT = 4096
+    if hard >= DEFAULT_SOFT:
+        soft = DEFAULT_SOFT
+
+    old_soft, old_hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+    hard = old_hard
+    if old_soft < soft:
+        if hard < soft:
+            hard = soft
+        resource.setrlimit(resource.RLIMIT_NOFILE, (soft, hard))
```

Comment:
I find this a little difficult to understand.  I think the double calls to `getrlimit` are part of the confusion.  Perhaps a comment as to the goal would be helpful.  It appears to ensure a minimal soft limit of `DEFAULT_SOFT` if the current hard limit is at least that much.

It seems like the `hard < soft` (L30) condition will never be true because `old_soft < soft` (L29) can only be true if `soft` was increased to `DEFAULT_SOFT`, which implies `hard >= soft`, which leads me to wonder some kind of update to `hard` is missing.

Reply:
Cleaned up